src/platform/w3dengine/client/w3ddisplay.h

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file
+ *
+ * @author Duncans_pumpkin
+ *
+ * @brief W3D class for the display handling.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+
+#include "coord.h"
+#include "display.h"
+#include "light.h"
+#include "render2d.h"
+#include "scene.h"
+
+class W3DAssetManager;
+
+class W3DDisplay final : public Display
+{
+public:
+    W3DDisplay();
+    ~W3DDisplay();
+    virtual void Init() override;
+    virtual void Reset() override;
+    virtual void Draw() override;
+    virtual void Set_Width(uint32_t width) override;
+    virtual void Set_Height(uint32_t height) override;
+    virtual bool Set_Display_Mode(uint32_t width, uint32_t height, uint32_t bits, bool windowed) override;
+    virtual int Get_Display_Mode_Count() override;
+    virtual void Get_Display_Mode_Description(int id, int *width, int *height, int *bit_depth) override;
+    virtual void Do_Smart_Asset_Purge_And_Preload(const char *) override;
```

Comment:
after Do_Smart_Asset_Purge_And_Preload add a comment that there's a virtual in WB here